packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,642 @@
+import {
+  CommittedTransactionInfo,
+  GatewayApiClient,
+  GatewayStatusResponse,
+  PublicKey,
+  StateEntityDetailsRequest,
+  StateEntityDetailsResponse,
+  TransactionCommittedDetailsRequest,
+  TransactionCommittedDetailsResponse,
+  TransactionPreviewRequest,
+  TransactionPreviewResponse,
+} from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  Convert,
+  Curve,
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  TransactionBuilder,
+  TransactionManifest,
+  generateRandomNonce,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+  singleFee,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, baseAmount } from '@xchainjs/xchain-util'
+import { bech32m } from 'bech32'
+import BIP32Factory, { BIP32Interface } from 'bip32'
+import { derivePath } from 'ed25519-hd-key'
+import * as ecc from 'tiny-secp256k1'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import {
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STOKENET_GATEWAY_URL,
+  XRD_DECIMAL,
+  XrdAsset,
+  bech32Networks,
+  xrdRootDerivationPaths,
+} from './const'
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  gatewayApiClient: GatewayApiClient
+  curve: Curve
+  constructor(
+    {
+      network = Network.Mainnet,
+      phrase,
+      rootDerivationPaths = xrdRootDerivationPaths,
+      feeBounds = {
+        lower: 1,
+        upper: 1,
+      },
+    }: XChainClientParams,
+    curve: Curve,
+  ) {
+    super(RadixChain, {
+      network: network,
+      phrase: phrase,
+      rootDerivationPaths: rootDerivationPaths,
+      feeBounds: feeBounds,
+    })
+    this.curve = curve
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    try {
+      const txParams: TxParams = {
+        asset: XrdAsset,
+        amount: baseAmount(1),
+        recipient: 'account_rdx1685t40mreptjhs9g3pg9lgf7k7rgppzjeknjgrpc7d0sumcjrsw6kj',
+      }
+
+      this.curve
+
+      const publicKey: PublicKey = {
+        key_hex: this.getRadixPrivateKey().publicKey().hexString(),
+        key_type: this.curve === 'Secp256k1' ? 'EcdsaSecp256k1' : 'EddsaEd25519',
+      }
+      const txManifest = this.prepareTx(txParams)
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const transactionPreviewRequest: TransactionPreviewRequest = {
+        manifest: (await txManifest).rawUnsignedTx,
```

Comment:
This would not work. This expects a manifest string while this is a hex string of an encoded manifest. We must decode the manifest first and use the instructions.

Reply:
Also, this should've been caught by tests. Do we have a test for `getFees`?

Reply:
I think this is fixed by your refactor